### PR TITLE
Parallax sample update and bug fix for multiple parallax items on page.

### PR DESCRIPTION
**Pull request by aghassemi**
Images are from pixabay.com and free for any use or modification under  Creative Commons CC0.. https://pixabay.com/en/service/faq/ 

Closes https://github.com/ampproject/amphtml/issues/7926 and https://github.com/ampproject/amphtml/issues/7925

### CODE REVIEW

File: examples/article-parallax.amp.html [hunk 1]
```diff
@@ -7,79 +7,161 @@
...
+        felis aliquam tortor vulputate, ac posuere velit semper.
+      </p>
+      <div class="sky-scene">
+        <amp-img width="1920" height="1280" layout="responsive" src="img/sky.jpg"></amp-img>
+        <div class="ballons">
+          <div class="ballon b1" amp-fx-parallax=1.3>
+            <amp-img layout="fill" src="img/ballon1.png"></amp-img>
+          </div>
+          <div class="ballon b2" amp-fx-parallax=1.15>
+            <amp-img layout="fill" src="img/ballon3.png"></amp-img>
```
Comment (muxin) [inline examples/article-parallax.amp.html]: Could we resolve this mismatch of numbers?

File: src/service/parallax-impl.js [hunk 2]
```diff
@@ -35,12 +34,6 @@ export class ParallaxService {
    * @param {!./ampdoc-impl.AmpDoc} ampdoc
    */
   constructor(ampdoc) {
-    /** @private @const {!Observable} */
-    this.parallaxObservable_ = new Observable();
```
Comment (cvializ) [inline src/service/parallax-impl.js]: I noticed you removed this listener from the `ParallaxService`, but didn't add something similar to the `ParallaxElement` class. You should still provide a way to detect scroll events to use in [`extensions/amp-fx-parallax/0.1/test/test-amp-fx-parallax.js`](https://github.com/aghassemi/amphtml/blob/869dd1ca1e563769f7c6701d699867d00a8819ab/extensions/amp-fx-parallax/0.1/test/test-amp-fx-parallax.js). I think the current PR will break that test.

File: src/service/parallax-impl.js [hunk 3]
```diff
@@ -187,9 +136,7 @@ export class ParallaxElement {
    */
   shouldUpdate(viewport) {
     const viewportRect = viewport.getRect();
-    const elementRect = viewport.getLayoutRect(this.element_);
-    elementRect.top -= viewportRect.top;
-    elementRect.bottom = elementRect.top + elementRect.height;
+    const elementRect = this.element_/*OK*/.getBoundingClientRect();
```
Comment (cvializ) [inline src/service/parallax-impl.js]: Why is it ok to use `getBoundingClientRect` here? Could you help me understand the purpose of the rule that disallows it?

File: src/service/parallax-impl.js [hunk 4]
```diff
@@ -168,16 +109,24 @@ export class ParallaxElement {
...
   }
 
   /**
    * Apply the parallax effect to the offset given how much the page
    * has moved since the last frame.
-   * @param {number} delta The movement of the base layer e.g. the page.
+   * @param {!./viewport-impl.Viewport} viewport.
    */
-  update(delta) {
+  update(viewport) {
```
Comment (cvializ) [inline src/service/parallax-impl.js]: Why pass a reference to the full viewport object when you only use `getScrollTop`? Consider passing the value from the calling function.

File: src/service/parallax-impl.js [hunk 5]
```diff
@@ -168,16 +109,24 @@ export class ParallaxElement {
 
     /** @private {number} */
     this.offset_ = 0;
+
+    /** @private {number} */
+    this.previousScroll_ = 0;
   }
 
   /**
    * Apply the parallax effect to the offset given how much the page
```
Comment (cvializ) [inline src/service/parallax-impl.js]: Update the description after "given", since now the function isn't given the delta.

**Comment by aghassemi** [comment]
@muxin @cvializ  Let's ignore this PR for now. The right solution for what I am trying to do requires a `parallax-scene` component, need to do more work, no point holding Canary. If we do another cut AMP Conf, that may get it.

File: src/service/parallax-impl.js (hunk 3 above)
Comment (jridgewell) [inline src/service/parallax-impl.js]: EEK! This is a horrible performance trap! We should be using `#getLayoutBox` when possible.

File: src/service/parallax-impl.js (hunk 4 above)
Comment (jridgewell) [inline src/service/parallax-impl.js]: > Why pass a reference to the full viewport object when you only use getScrollTop?
👍

**Comment by aghassemi** [comment]
further work needs to be part of scroll-bound animation service. closing this (which was suppose to be a stop gap for amp conf demo)

**Pull request closed without merging**